fix(transfers): Added implementations for TransferItem and Account mappers and MoneyTransferItemsRepository, fixes failing tests

mj-shifu opened the pull request.

mj-shifu committed:
fix(transfers)

- implemented mappers for TransferItems and Accounts
- implemented MoneyTransferItemsRepository

mj-shifu committed:
Merge branch 'NilsWild:main' into main

NilsWild commented on piggybank-transfers-spring-rest/src/main/kotlin/de/rwth/swc/piggybank/transfers/repository/entity/mapping/MoneyTransferItemMapper.kt:
```diff
@@ -29,15 +36,23 @@ interface MoneyTransferItemMapper {
+        return MoneyTransferItemEntity(
+            id = item.id.value,
+            amount = item.amount.amount.value,
+            currency = item.amount.currency.to(),
```
to is not selfexplanatory. the method should be renamed

NilsWild commented on piggybank-transfers-spring-rest/src/main/kotlin/de/rwth/swc/piggybank/transfers/repository/entity/mapping/MoneyTransferItemMapper.kt:
```diff
@@ -47,6 +62,16 @@ class MoneyTransferItemMapperImpl : MoneyTransferItemMapper {
+            id = MoneyTransferItemId(entity.id),
+            amount = Money(
+                amount = MoneyAmount(entity.amount),
+                currency = Currency.from(entity.currency),
```
from is not selefexplanatory. what string should be passed there? e.g. rename the method to fromISOCode

NilsWild commented on piggybank-transfers-spring-rest/src/main/kotlin/de/rwth/swc/piggybank/transfers/repository/entity/mapping/AccountMapper.kt:
```diff
@@ -0,0 +1,69 @@
+     */
+    override fun toPersistence(item: Account): AccountEntity {
+        return AccountEntity(
+            id = Generators.nameBasedGenerator().generate("id-"+item.type + "-" + item.identifier),
```
it should be tested that different Accounts with the same value get mapped to the same entity

NilsWild reviewed (changes requested):
address my comments on your changes to get the pull request approved

mj-shifu committed:
feat(transfers): test coverage and refactoring

- added tests for AccountMapper
- to and from ISOCode are now called toISOCode and fromISOCode

NilsWild commented on piggybank-domain/src/main/kotlin/de/rwth/swc/piggybank/domain/shared/valueobject/Currency.kt:
```diff
@@ -25,9 +25,31 @@ data class Currency(
          * @return A new instance of Currency.
          */
-        fun from(name: String, symbol: String, decimalPlaces: Int): Currency {
+        fun fromISOCode(name: String, symbol: String, decimalPlaces: Int): Currency {
```
this is not from ISO Code. Probably keep it as "from" as it uses all the atomic attributes. However we should rename name to isoCode and eventually do the same to the classes attribute

NilsWild commented on piggybank-domain/src/main/kotlin/de/rwth/swc/piggybank/domain/shared/valueobject/Currency.kt:
```diff
@@ -25,9 +25,31 @@ data class Currency(
             return Currency(CurrencyName(name), CurrencySymbol(symbol), DecimalPlaces(decimalPlaces))
         }
+
+        fun fromISOCode(id: String): Currency {
```
rename parameter from id to iso code

NilsWild commented on piggybank-transfers-spring-rest/pom.xml:
```diff
@@ -105,6 +105,11 @@
             <groupId>de.rwth.swc</groupId>
             <artifactId>piggybank-domain</artifactId>
         </dependency>
+
```
this is now a duplicate  remove

NilsWild commented on piggybank-transfers-spring-rest/src/test/kotlin/de/rwth/swc/piggybank/transfers/repository/entity/mapping/AccountMapperTest.kt:
```diff
@@ -0,0 +1,88 @@
+
+        val persistentAccount1 = mapper.toPersistence(account1)
+        val persistentAccount2 = mapper.toPersistence(account2)
+        persistentAccount1.type shouldBe persistentAccount2.type
```
just compare them with shouldBe. no need to check individual attributes. Especially if additional fields are added later we want them to be checked automatically

NilsWild commented on piggybank-transfers-spring-rest/src/test/kotlin/de/rwth/swc/piggybank/transfers/repository/entity/mapping/AccountMapperTest.kt:
```diff
@@ -0,0 +1,88 @@
+    lateinit var mapper: AccountMapper
+
+    @Test
+    fun `same acccounts are mapped to same id`() {
```
same accounts are mapped to same persistence entity

mj-shifu committed:
fix(transfers): rename fromISOCode to from

Syndicatom committed:
fix: changed Money to String Formating to always use German Local. (#1)

mj-shifu committed:
feat(transfers): tests for fromISOCode

- renamed AccountMapper tests
- fromISOCode is now based on reflection
- added test for fromISOCode

mj-shifu committed:
fix(transfers): remove duplicate from pom.xml

mj-shifu committed:
fix(transfers): minor fixes

- fix unresolved reference
- simplify test checks
- run tests on whole project

mj-shifu committed:
fix(transfers): simplify reflection for getISOCode

mj-shifu committed:
fix(transfers): fix pom.xml

mj-shifu committed:
Merge branch 'NilsWild:main' into main
